src/metabase/driver/sql_jdbc/connection.clj

```diff
@@ -157,6 +165,23 @@
                     ;; don't log the actual spec lest we accidentally expose credentials
                     {:input (class db-or-id-or-spec)}))))
 
+(defn invalidate-pool-if-ssh-tunnel-closed!
+  "If the pool for the given database uses an ssh tunnel, and that tunnel is closed, then close the pool and
+   remove it from the cache. This will force the next call to db->pooled-connection-spec to reopen the tunnel."
+  [database]
+  (let [conn-spec (db->pooled-connection-spec database)]
+    (if
```

Comment:
this should be a `when` which has an implicit `do`:

Reply:
```clojure
(when (and (ssh/use-ssh-tunnel? conn-spec)
               (not (ssh/ssh-tunnel-open? conn-spec)))
      (log/warn
       (u/format-color
           'red
           (trs "ssh tunnel for database {0} appears to have closed; closing the pool so it can be reestablished"
                (u/the-id database))))
      (invalidate-pool-for-db! database))
```

Reply:
Fixed.